azerothcore/azerothcore-wotlk
# fix(DB/Quest): Fixed some of the Stave of the Ancients quest

## Changes Proposed:
-  Added scripting, fixed waypoints and added gossip options to the 4 demons

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/6966

Used the scripting provided on https://github.com/TrinityCore/TrinityCore/issues/22398 with some changes to be able to update in our database

## SOURCE:

Information about the quest from here:
https://www.youtube.com/watch?v=adz-VM7wzo4

## Tests Performed:

- .go c 14527, 14529, 14531, 14536
- Check if the gossip works.
- Check if they move
- Check if they fight like they should

Small video

https://streamable.com/57dksm

## How to Test the Changes:


1. .go c 14527, 14529, 14531, 14536
2.  Check if the gossip works.
3.  Check if they move
4.  Check if they fight like they should

## Known Issues and TODO List:


- The script npc_stave_of_the_ancientsAI break this quest. Im still need to check it, but im more of a Java guys, so i might need somone to check it for me.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
